### PR TITLE
fix: close Select's dropdowns

### PR DESCRIPTION
**Pull request by dozortsev**
## 🖼 Context

<details>
  <summary>Atm it's possible to open multiple Select's dropdowns which is considered as buggy behaviour, examples: </summary>
  
image

Screen Shot 2021-02-05 at 11 55 48 AM

</details>

The above issue was introduced in fix https://github.com/transferwise/neptune-web/pull/572 for another thing in Select, but contributor was having that issue because he was using component with React 17 ([I haven't confirmed this from him](https://github.com/transferwise/neptune-web/pull/753#issuecomment-776862708) but I was able to reproduce this issue my self by setup latest CRA), where React introduced breaking changes around [events delegation](https://reactjs.org/blog/2020/10/20/react-v17.html#changes-to-event-delegation) ([that is also why we couldn't reproduce this bug in our docs website nor on our Storybook as both rely on React 16](https://github.com/transferwise/neptune-web/issues/752#issuecomment-751363963)).

Atm I couldn't find a way how to make this component work with both versions of React (I did try follow suggestion from [React docs](https://reactjs.org/blog/2020/08/10/react-v17-rc.html#fixing-potential-issues) and some other [examples](https://stackoverflow.com/questions/7398290/unable-to-understand-usecapture-parameter-in-addeventlistener/7398447#7398447)), but it's either works correct on React 17 but doesn't work on 16 and visa-versa, not sure it's even possible, perhaps major refactoring/whole rewrite would help.
 
## 🚀 Changes

Simple revert the change made in https://github.com/transferwise/neptune-web/pull/572

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/select/Select.js [hunk 1]
```diff
@@ -270,8 +270,8 @@ export default class Select extends Component {
     document.removeEventListener('click', this.handleDocumentClick, false);
   }
 
-  handleButtonClick = (e) => {
-    stopPropagation(e);
+  handleButtonClick = () => {
+    // stopPropagation(e);
```
Comment (dozortsev) [inline packages/components/src/select/Select.js:274]: revering fix https://github.com/transferwise/neptune-web/pull/753 which introduced another bug

**Review by helendurranttw — COMMENTED** [review]
This looks fine to me, though you should check to see you don't re-introduce the bug that the change you reverted was trying to fix 😅

**Comment by dozortsev** [comment]
yep, in this PR I will try fix the issue https://github.com/transferwise/neptune-web/pull/753 just in another way

File: packages/docs/liveEditorCode/select.code.js [hunk 2]
```diff
@@ -49,9 +37,6 @@
         {
           value: 8,
           label: 'A thing with searchable alternatives',
-          /**
-           * searchStrings is an array of strings that maps additional searchable terms to a given option.
-           */
```
Comment (dozortsev) [inline packages/docs/liveEditorCode/select.code.js:54]: deleting redundant comments from docs

**Review by helendurranttw — APPROVED** [review]

**Pull request merged**